### ISSUE TITLE
JUnit5 not working (in Eclipse)

### ISSUE THREAD

**Issue by hohwille** [devonfw/devon4j#166]
### Expected behavior

As a developer, I want to write and run JUnit tests so that I can assure quality.

### Actual behavior

When I run my JUnits in Eclipse, I get this error:

```
java.lang.NoSuchMethodError: org.junit.platform.launcher.Launcher.execute(Lorg/junit/platform/launcher/TestPlan;[Lorg/junit/platform/launcher/TestExecutionListener;)V
```
See also here:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=545849

Seems this was never tested before releasing. Is Eclipse not used for MTS(J)?

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1.  Create devon4j app from 3.2.0 app template
2. Launch JUnit in Eclipse

### Related/Dependent Issues
See #151 

### Comments/Hints:
IMHO the problem is that `devon4j-test` comes with dependencies such as `junit-platform-runner` and `junit-platform-launcher` in an obsolete version `1.3.2`. These dependencies should either not be shipped with devonfw libs if that is possible or we need to ship a recent version (at least `1.4.2`)

### Affected version:
`devon4j:3.2.0`

**Comment by hohwille** [comment]
due to the urgency, I am taking over this issue.

**Comment by hohwille** [comment]
We are including spring-boot BOM in version `2.1.9.RELEASE` which already includes junit in a wrong/broken version:
```
    <junit-jupiter.version>5.3.2</junit-jupiter.version>
```

**Comment by hohwille** [comment]
I got this working by upgrading to spring-boot `2.0.0` but now `@Inject` is not working into JUnits so `DependencyInjectionTestExecutionListener` stopped working.
Seems that JUnit5 is together with its ecosystem still in some kind of experimental state and we migrated too early?

**Comment by hohwille** [comment]
Seems I got the cause why `DependencyInjectionTestExecutionListener` stopped working. `@SpringJUnitConfig` is the magic link that was missing (technically `@ContextConfiguration`):
https://docs.spring.io/spring/docs/current/spring-framework-reference/testing.html#integration-testing-annotations-junit-jupiter

**Comment by hohwille** [comment]
Fixed with PR #168
